### PR TITLE
Staging for version increment automation

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
This is the staging PR for to execute workflows that auto raise the version increment PR's
Example: https://github.com/prudhvigodithi/dashboards-reports/pull/3

### Issues Resolved
Part of: https://github.com/opensearch-project/opensearch-build/issues/1375
From solution: https://github.com/opensearch-project/opensearch-build/issues/1375#issuecomment-1159476686
Related issue: https://github.com/opensearch-project/dashboards-reports/issues/385

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by prudhvigodithi** [comment]
Referred old version increment PR.
https://github.com/opensearch-project/dashboards-reports/pull/371/files

**Review by joshuali925 — APPROVED** [review]
LGTM. questions:

1. since this is mostly regex matching, any reason to use a gradle task instead of more generic solutions (python/shell scripts)? Some repos are not java/gradle projects
2. When is this task triggered?

Thanks

**Review by kavithacm — DISMISSED** [review]

**Comment by prudhvigodithi** [comment]
Renaming to task updateVersion following https://github.com/opensearch-project/opensearch-plugin-template-java/pull/32.

**Comment by prudhvigodithi** [comment]
Hey @anirudha @joshuali925, if we are ok, can we merge this PR.
Thank you

**Comment by prudhvigodithi** [comment]
> LGTM. questions:
> 
> 1. since this is mostly regex matching, any reason to use a gradle task instead of more generic solutions (python/shell scripts)? Some repos are not java/gradle projects
> 2. When is this task triggered?
> 
> Thanks

Hey @joshuali925 gradle is better as the solution would be generic across for all repos, a script would be not as generic and would need to be configured one per plugin and soon will be hard to manage, this gradle task `updateVersion` will be called using [workflow](https://github.com/opensearch-project/opensearch-build/blob/main/.github/workflows/increment-plugin-versions.yml) from build repo.
Solution details: https://github.com/opensearch-project/opensearch-build/issues/1375#issuecomment-1159476686

File: reports-scheduler/build.gradle [hunk 1]
```diff
@@ -559,3 +559,21 @@ if (!usingRemoteCluster && !usingMultiNode) {
...
 apply from: 'build-tools/pkgbuild.gradle'
+
+// versionIncrement: Task to auto increment to the next development iteration
+task versionIncrement {
+    onlyIf { System.getProperty('newVersion') }
+    doLast {
+        ext.newVersion = System.getProperty('newVersion')
+        println "Setting version to ${newVersion}."
+         // String tokenization to support -SNAPSHOT
+        ant.replaceregexp(file:'build.gradle', match: '"opensearch.version", "\\d.*"', replace: '"opensearch.version", "' + newVersion.tokenize('-')[0] + '-SNAPSHOT"', flags:'g', byline:true)
```
[the remark this answers is not in the document]
Comment (prudhvigodithi) [reply reports-scheduler/build.gradle:570]: Yes the default value is always added as `-SNAPSHOT`, however this is a run time configured value and would be modified by `build.sh`
https://github.com/opensearch-project/dashboards-reports/blob/main/reports-scheduler/scripts/build.sh#L71

**Review by joshuali925 — APPROVED** [review]

**Review by kavithacm — APPROVED** [review]

**Pull request merged**